### PR TITLE
Fix for Cannot view newly created flux 2 package from UX #4173 

### PR DESCRIPTION
**Pull request by gfichtenholt**

### CODE REVIEW

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go [hunk 1]
```diff
@@ -676,3 +665,24 @@ func installedPackageAvailablePackageRef(rel *helmv2.HelmRelease) (*corev1.Avail
...
 }
+
+// ref https://fluxcd.io/docs/components/helm/helmreleases/
+func helmReleaseName(key types.NamespacedName, rel *helmv2.HelmRelease) types.NamespacedName {
+	helmReleaseName := rel.Spec.ReleaseName
+	// according to docs ReleaseName is optional and defaults to a composition of
+	// '[TargetNamespace-]Name'.
+	if helmReleaseName == "" {
+		targetNamespace := rel.Spec.TargetNamespace
+		// according to docs targetNamespace is optional and defaults to the namespace of the HelmRelease
```
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go:676]: Just noting, not necessary for this fix, but for the related #4189, that the behaviour flux shows is slightly different to below (see my notes from updating the flux CR on #4172, or your notes on #4189) but still consistent with the comment above: the namespace targeted by the `HelmRelease` defaults to the namespace of the `HelmRelease`, but flux doesn't set the `targetNamespace` field with that value, it remains empty as specified. And when the `targetNamespace` is empty/unset, the helm release name is just the `HelmRelease.metadata.name` (ie. not prefixed by the namespace at all, and so different to below). This is why I was unsure why we were explicitly setting the `targetNamespace` given flux's default behaviour.

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go [hunk 2]
```diff
@@ -676,3 +665,24 @@ func installedPackageAvailablePackageRef(rel *helmv2.HelmRelease) (*corev1.Avail
...
+		if targetNamespace == "" {
+			targetNamespace = key.Namespace
+		}
+		helmReleaseName = fmt.Sprintf("%s-%s", targetNamespace, key.Name)
```
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go:680]: Just leaving this here so I can refer to it on your related issue #4189, not suggesting the change for this PR

```suggestion
		helmReleaseName = key.Name
		if targetNamespace != "" {
			helmReleaseName = fmt.Sprintf("%s-%s", targetNamespace, key.Name)
		}
```
which would match the behaviour that I saw by default from flux, I think.

**Review by absoludity — APPROVED** [review]
Excellent, Greg, thanks for the quick fix!

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go (hunk 1 above)
Comment (gfichtenholt) [reply cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go:676]: > Just noting, not necessary for this fix, but for the related #4189, that the behaviour flux shows is slightly different to below (see my notes from updating the flux CR on #4172, or your notes on #4189) but still consistent with the comment above: the namespace targeted by the `HelmRelease` defaults to the namespace of the `HelmRelease`, but flux doesn't set the `targetNamespace` field with that value, it remains empty as specified. And when the `targetNamespace` is empty/unset, the helm release name is just the `HelmRelease.metadata.name` (ie. not prefixed by the namespace at all, and so different to below). This is why I was unsure why we were explicitly setting the `targetNamespace` given flux's default behaviour.

noted

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go (hunk 2 above)
Comment (gfichtenholt) [reply cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go:680]: ok, sounds good

**Pull request merged**